### DISCUSSION TITLE
Remove (unused) dependency from API function

### DISCUSSION THREAD

**Discussion by michaelwitk** [vercel/next.js#35455] [Help]
I use https://www.npmjs.com/package/fabric on my frontend (no SSR). To get it to work on Vercel I had to follow https://github.com/Automattic/node-canvas/issues/1779 . So far so good, but apparently my api functions take up a lot of size and I've hit the cap just now. https://vercel.com/docs/concepts/limits/overview#serverless-function-size

```
Serverless Function's page: api/post.js
16:32:27.136 | Large Dependencies                 Uncompressed size  Compressed size
16:32:27.137 | node_modules/canvas/build                     172 MB          44.9 MB
16:32:27.137...
```

Since I don't need `node_modules/canvas/build` may I exclude it? This does not seem to do the trick:

```
// vercel.json
{
    "functions": {
      "pages/api/**/*.js": {
        "excludeFiles": "{node_modules/canvas/**}"
      }
    }
  }
  ```

Some links that may be useful for helping me out:
https://vercel.com/docs/cli#project-configuration/functions
https://github.com/isaacs/node-glob#glob-primer
 

edit: 
ended up dynamically importing `fabric` which solves my issue, albeit a solution would be neat 
```jsx
import dynamic from "next/dynamic"

export default dynamic(() => import('./FabricComponent'), { ssr: false })
```

**Comment by Siva2406** [comment]
Hey @michaelwitk, We are also facing a similar issue on our project. Were you able to resolve this issue? Any assistance you can provide would be greatly appreciated.
